integrations/docker/images/chip-cert-bins/Dockerfile

```diff
@@ -0,0 +1,198 @@
+            && rm -rf cmake-3.19.3-Linux-x86_64.sh) \
+            && exec bash \
+            ;; \
+        "linux/arm64") \
+            set -x \
+            && (cd /tmp \
+            && wget --progress=dot:giga https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-aarch64.sh \
+            && sh cmake-3.19.3-Linux-aarch64.sh --exclude-subdir --prefix=/usr/local \
+            && rm -rf cmake-3.19.3-Linux-aarch64.sh) \
+            && exec bash \
+            ;; \
+        *) ;; \
+    esac
+
+# Python 3.9 and PIP
+RUN set -x \
+    && DEBIAN_FRONTEND=noninteractive  apt-get update \
+    && DEBIAN_FRONTEND=noninteractive  apt-get install -y libgirepository1.0-dev \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common \
+    && add-apt-repository universe \
+    && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && python3.9 get-pip.py \
+    && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1 \
+    && rm -rf /var/lib/apt/lists/ \
+    && : # last line
+
+RUN set -x \
+    && pip3 install attrs coloredlogs PyGithub pygit future portpicker mobly click cxxfilt ghapi pandas tabulate \
+    && : # last line
+
+# build and install gn
+RUN set -x \
+    && git clone https://gn.googlesource.com/gn \
+    && cd gn \
+    && python3 build/gen.py \
+    && ninja -C out \
+    && cp out/gn /usr/local/bin \
+    && cd .. \
+    && rm -rf gn \
+    && : # last line
+
+# Install bloat comparison tools
+RUN set -x \
+    && git clone https://github.com/google/bloaty.git \
+    && mkdir -p bloaty/build \
+    && cd bloaty/build \
+    && cmake ../ \
+    && make -j8 \
+    && make install \
+    && cd ../.. \
+    && rm -rf bloaty \
+    && : # last line
+
+# NodeJS: install a newer version than what apt-get would read
+# This installs the latest LTS version of nodejs
+RUN case ${TARGETPLATFORM} in \
+        "linux/amd64") \
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-x64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-x64.tar.xz \
+			&& mv node-v12.19.0-linux-x64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-x64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        "linux/arm64")\
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-arm64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-arm64.tar.xz \
+			&& mv node-v12.19.0-linux-arm64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-arm64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        *) ;; \
+    esac
+
+# Set python to python3
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+RUN mkdir /root/connectedhomeip
+ADD ./ /root/connectedhomeip/
+WORKDIR /root/connectedhomeip/
+RUN scripts/build/gn_bootstrap.sh
```

Comment:
what version of code will this pick?

My concern is that we currently version docker images and generally expect that a build of docker image a.b.c is always with the same content regardless of when it is built (so we can rebuild them if need be). 

This seems to be "pick current checkout"  so content will presumably vary on every single checkout sha.

Reply:
Yes it will use the currently checked-out version, along with any local modifications. I did this so that there can be flexibility for the builder in case custom changes need to be made for a specific test harness use.

We can definitely write a wrapper script which will fetch and check out origin/HEAD for a CI use case.

Reply:
Please use a SHA. Having every dockerfile be a moving target is not ok (especially the 'pick local modifications' part)